# GCC 7 errors on comparing a pointer and an integer, code means to test
# first byte. This is fixed in 6.4.0 and onward by SVN revision 239971.
# Affects 6.1-6.3.
# Upstream credit: Kirill Yukhin
diff -ru gcc-6.3.0-orig/gcc/ubsan.c gcc-6.3.0/gcc/ubsan.c
--- gcc-6.3.0-orig/gcc/ubsan.c	2016-12-07 23:39:29.000000000 +0100
+++ gcc-6.3.0/gcc/ubsan.c	2017-07-05 08:57:03.297566616 +0200
@@ -1471,7 +1471,7 @@
 
   expanded_location xloc = expand_location (loc);
   if (xloc.file == NULL || strncmp (xloc.file, "\1", 2) == 0
-      || xloc.file == '\0' || xloc.file[0] == '\xff'
+      || xloc.file[0] == '\0' || xloc.file[0] == '\xff'
       || xloc.file[1] == '\xff')
     return false;
 
